content/general/concepts/data-structures/terms/algorithms/searching/terms/binary-search/binary-search.md

```diff
@@ -0,0 +1,69 @@
+---
+Title: 'Binary Search Algorithm'
+Description: 'A searching algorithm to search an element in a sorted collection by dividing the collection in two halves.'
+Subjects:
+  - 'Computer Science'
+  - 'Code Foundations'
+Tags:
+  - 'Algorithms'
+  - 'Data Structures'
+  - 'Search Algorithms'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+**Binary Search** is an algorithm for searching an element within a sorted collection of items, primarily implemented with [arrays](https://www.codecademy.com/resources/docs/general/data-structures/array) or lists. The binary search algorithm follows a divide-and-conquer approach to determine whether the target element matches the collection's middle element. If there is a match, it provides the index of the middle element; otherwise, it proceeds to either side of the array, depending on the current comparison result.
+
+> **Note**: The collection must be sorted for the implementation of the binary search algorithm.
+
+## The Algorithm
+
+The steps for the binary search algorithm are as follows: 
+
+1. Set the `start` pointer to the beginning of the collection (index 0).
+2. Set the `end` pointer to the end of the collection (length(collection) - 1).
+3. While the `start` is less than or equal to the `end` pointer, repeat these steps:
+    1. Calculate the middle element index: `mid = start + (end - start) / 2`.
```

Comment:
Very good that you showed `mid = start + (end - start)/2` as opposed to `mid = (start + end)/2`

Reply:
Yup, as `mid = start + (end - start)/2` is more efficient to use than `mid = (start + end)/2`.